src/app/view/perspectives/calendar/config/calendar-config.component.ts

```diff
@@ -75,4 +87,32 @@ export class CalendarConfigComponent {
 
     this.configChange.emit(config);
   }
+
+  private getSlotDurationItems(): SelectItemModel[] {
+    return Object.keys(SlotDuration).map(key => {
+      return {
+        id: key,
+        value: this.getDurationString(key),
+      };
+    });
+  }
+
+  private getDurationString(key: string) {
+    return this.i18n(
+      {
+        id: 'perspective.calendar.config.slotDuration.value',
+        value: '{key, select, Hour {1 hour} Half {30 minutes} Quarter {15 minutes} Ten {10 minutes} Five {5 minutes}}',
+      },
+      {
+        key,
+      }
+    );
+  }
+
+  public onSlotDurationChange(selected: SlotDuration) {
+    const config = deepObjectCopy<CalendarConfig>(this.config);
```

Comment:
You don't need to make deep copy in this situation. You can use shallow copy `const config = {...this.config, slotDuration, selected}`